### PR TITLE
Revolvers can now ricochet off of coins

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

Laser/energy projectiles and the detective's revolver already did this so I made the other revolvers do it as well. Also, throwing a coin will now flip the coin.

# Changelog

:cl:  
tweak: all revolvers can ricochet off of coins, ultra violence version can do so on the hit that causes the coin to flip
tweak: coins now flip when thrown
tweak: ricochet now prefers closer targets
tweak: get_closest_atom() now picks randomly from a list of closest atoms if there are multiple closest atoms with the same distance
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
tested?

**Comment by iamthesilly** [comment]
> tested?

Yes

File: code/modules/mining/ores_coins.dm [hunk 1]
```diff
@@ -529,10 +531,10 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 	transform = initial(transform)
 
 /obj/item/coin/bullet_act(obj/item/projectile/P)
-	if(P.flag != LASER && P.flag != ENERGY && !istype(P, /obj/item/projectile/bullet/c38)) //only energy projectiles get deflected (also det revolver because damn thats cool)
+	if(P.flag != LASER && P.flag != ENERGY && !istype(P, /obj/item/projectile/bullet/c38) && !istype(P, /obj/item/projectile/bullet/ipcmartial)) //only energy projectiles get deflected (also det and IPC revolvers because damn thats cool)
 		return ..()
 		
-	if(cooldown >= world.time)//we ricochet the projectile
+	if(cooldown >= world.time || istype(P, /obj/item/projectile/bullet/ipcmartial))//we ricochet the projectile
```
Comment (ynot01) [inline code/modules/mining/ores_coins.dm:537]: document the cooldown bypass

Comment (iamthesilly) [reply code/modules/mining/ores_coins.dm:537]: done

**Comment by Moltijoe** [comment]
you've tested the new changes to get_closest_atom?

**Comment by iamthesilly** [comment]
> you've tested the new changes to get_closest_atom?

Yes, works as intended

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**